javaslang/src/main/java/javaslang/Value.java

```diff
@@ -516,7 +516,7 @@ default CharSeq toCharSeq() {
      */
     @SuppressWarnings("unchecked")
     @GwtIncompatible("reflection is not supported")
-    default T[] toJavaArray(Class<T> componentType) {
+    default T[] toJavaArray(Class<? super T> componentType) {
```

Comment:
I think some unsafety is inherent. Arrays are really messy.
Since I find it unlikely to occur in real life (the primitive one should occur a lot more often), and since it will fail every time, signaling a real error (though not compile-time, but run-time), I think it's low-risk, and useful.